pkg/blockbuilder/scheduler/metrics.go

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package scheduler
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type schedulerMetrics struct {
+	monitorPartitionsDuration prometheus.Histogram
+	partitionStartOffsets     *prometheus.GaugeVec
+	partitionEndOffsets       *prometheus.GaugeVec
+}
+
+func newSchedulerMetrics(reg prometheus.Registerer) schedulerMetrics {
+	return schedulerMetrics{
+		monitorPartitionsDuration: promauto.With(reg).NewHistogram(prometheus.HistogramOpts{
```

Comment:
In this PR at the moment this metric measures the time it takes to fetch the offsets, but what's your longer term idea? If you will just track the offsets fetching timing with this metric, then just make it explicit (e.g. `cortex_blockbuilder_scheduler_fetch_offsets_duration_seconds`) otherwise if you plan to track the entire scheduling I would already name it accordingly (e.g. `cortex_blockbuilder_scheduler_scheduling_duration_seconds` ... or something better you can come up with).

Reply:
Ah, yeah. This method will do all of:

1. fetch all necessary quantities from Kafka to make scheduling decisions. (Still have a couple more to add.)
2. update the schedule

Number 1 is the interesting part, timing-wise, but the timing covers both. Number 2 is going to be microseconds. To tie in your other good comment, this method will be renamed to something more like updateSchedule and the metric can be paired as `scheduling_duration`.